src/Chapter2.hs

```diff
@@ -728,7 +735,8 @@ value of the element itself
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate [] = []
+smartReplicate l = concatMap (\x -> replicate x x) l
```

Comment:
Great implementation! You don't need to handle an empty list as a separate pattern. Also, you can eta-reduce the implementation a bit :slightly_smiling_face: